src/main/java/com/synopsys/blackduck/upload/client/UploaderConfig.java

```diff
@@ -267,47 +268,99 @@ private void validate() throws IntegrationException {
             }
         }
 
-        private int getUploadChunkSize() {
-            return Integer.parseInt(getPropertyValue(EnvironmentProperties.BLACKDUCK_UPLOAD_CHUNK_SIZE.getPropertyKey()));
+
+        /**
+         * Retrieve current builder value for the size of uploading chunks to Black Duck.
+         *
+         * @return configured or default upload chunk size.
+         */
+        public int getUploadChunkSize() {
+            Optional<String> uploadChunkSizeProperty = Optional.ofNullable(getPropertyValue(EnvironmentProperties.BLACKDUCK_UPLOAD_CHUNK_SIZE.getPropertyKey()));
+            return uploadChunkSizeProperty.map(Integer::parseInt).orElse(UploadValidator.DEFAULT_UPLOAD_CHUNK_SIZE);
         }
 
-        private int getTimeoutInSeconds() {
-            return Integer.parseInt(getPropertyValue(EnvironmentProperties.BLACKDUCK_TIMEOUT_SECONDS.getPropertyKey()));
+        /**
+         * Retrieve current builder value for the timeout when communicating with Black Duck.
+         *
+         * @return configured or default timeout in seconds.
+         */
+        public int getTimeoutInSeconds() {
+            Optional<String> timeoutInSeconds = Optional.ofNullable(getPropertyValue(EnvironmentProperties.BLACKDUCK_TIMEOUT_SECONDS.getPropertyKey()));
+            return timeoutInSeconds.map(Integer::parseInt).orElse(300);
```

Comment:
Have a DEFAULT_BLACKDUCK_TIMEOUT_SECONDS value in UploadValidator similar to the other values.